generator/internal/dart/dart.go

```diff
@@ -189,11 +189,25 @@ func httpPathArgs(_ *api.PathInfo) []string {
 }
 
 func formatDocComments(documentation string, _ *api.APIState) []string {
-	ss := strings.Split(documentation, "\n")
-	for i := range ss {
-		ss[i] = strings.TrimRightFunc(ss[i], unicode.IsSpace)
+	lines := strings.Split(documentation, "\n")
```

Comment:
I wonder if we should refactor this, because we do something similar in Rust.

Reply:
I think much of the doc processing could be shared, or parts shared w/ some parameterization. The last 10-20% would likely need to be backend specific (Dart doesn't require fenced code blocks to analyze correctly, has a different way of resolving code references in comments, ...).